contracts/zero-ex/contracts/src/features/OtcOrdersFeature.sol

```diff
@@ -179,97 +222,146 @@ contract OtcOrdersFeature is
         override
         returns (uint128 takerTokenFilledAmount, uint128 makerTokenFilledAmount)
     {
-        return _fillOtcOrderPrivate(
-            order,
+        LibNativeOrder.OtcOrderInfo memory orderInfo = getOtcOrderInfo(order);
+        address taker = LibSignature.getSignerOfHash(orderInfo.orderHash, takerSignature);
+        
+        _validateOtcOrder(
+            order, 
+            orderInfo,
             makerSignature,
-            takerSignature,
-            order.takerAmount,
-            unwrapWeth ? WethOptions.UnwrapWeth : WethOptions.LeaveAsWeth
+            taker
```

Comment:
note to self: taker address is recovered from takerSignature and validated against order field